prow/cmd/audit-owners/main.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This application reads kubernetes OWNERS files within a repo and compares it
+// to collaborators for a repo to find those listed in OWNERS files who are not
+// collaborators
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"k8s.io/test-infra/prow/github"
+)
+
+// The root location of the repo to start walking the filesystem
+var repoRoot string
+
+// The name of the GitHub org to use when querying for collaborators
+var githubOrg string
+
+// The name of the GitHub repo to use when querying for collaborators
+var githubRepo string
+
+func init() {
+	flag.StringVar(&repoRoot, "s", "", "The location of the repo on the local filesystem to start inspecting")
+	flag.StringVar(&githubOrg, "o", "", "The name of the GitHub org to use when querying for collaborators")
+	flag.StringVar(&githubRepo, "r", "", "The name of the GitHub repo to use when querying for collaborators")
+}
+
+func main() {
+
+	flag.Parse()
+
+	// We require a github token to make API calls due to rate limiting
+	if os.Getenv("GITHUB_TOKEN") == "" {
+		fmt.Println("Error: Please supply an environment variable named GITHUB_TOKEN with a valid token")
+		os.Exit(1)
+	}
+
+	// Ensure all the flags were passed in
+	if repoRoot == "" {
+		fmt.Println("Error: Please supply a local source code location with the -s flag")
+		os.Exit(1)
+	}
+	if githubOrg == "" {
+		fmt.Println("Error: Please supply a GitHub org with the -o flag")
+		os.Exit(1)
+	}
+	if githubRepo == "" {
+		fmt.Println("Error: Please supply a GitHub repo name with the -r flag")
+		os.Exit(1)
+	}
+
+	// Stores the usernames found in the OWNERS files
+	handles := make(map[string]struct{})
+
+	// Walk the file tree to gather all the GitHub Logins used in all OWNERS files
+	err := filepath.Walk(repoRoot, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		name := info.Name()
+
+		// Make sure to skip anything in the .git directory. OWNERS files can be
+		// found in there but their format will cause read errors.
+		if name == ".git" {
+			return filepath.SkipDir
+		}
+
+		// Add in handling for OWNERS_ALIASES files
```

Comment:
This is important. Without it most of the logins identified as not being collaborators will actually be aliases.